iceoryx_posh/source/roudi/port_pool.cpp

```diff
@@ -160,6 +160,52 @@ PortPool::addSubscriberPort(const capro::ServiceDescription& serviceDescription,
     }
 }
 
+cxx::vector<popo::ClientPortRouDi::MemberType_t*, MAX_CLIENTS> PortPool::getClientPortDataList() noexcept
+{
+    return m_portPoolData->m_clientPortMembers.content();
+}
+
+cxx::vector<popo::ServerPortRouDi::MemberType_t*, MAX_SERVERS> PortPool::getServerPortDataList() noexcept
+{
+    return m_portPoolData->m_serverPortMembers.content();
+}
+
+cxx::expected<popo::ClientPortRouDi::MemberType_t*, PortPoolError>
+PortPool::addClientPort(const capro::ServiceDescription& serviceDescription,
+                        mepoo::MemoryManager* const memoryManager,
+                        const RuntimeName_t& runtimeName,
+                        const popo::ClientOptions& clientOptions,
+                        const mepoo::MemoryInfo& memoryInfo) noexcept
+{
+    if (!m_portPoolData->m_clientPortMembers.hasFreeSpace())
+    {
+        errorHandler(Error::kPORT_POOL__CLIENTLIST_OVERFLOW, nullptr, ErrorLevel::MODERATE);
```

Comment:
Could you please add a `LogError` here with service description and runtime name.

Reply:
Good point. I C&P this code and will also add the log to the source location. I have to look at the error handling document if it must be `LogWarn` or `LogError`